### PR TITLE
dirac finance tvl infos

### PR DESCRIPTION
**Pull request by salimdev04**
Name (to be shown on DefiLlama): Dirac Finance
Twitter Link: https://twitter.com/DiracFinance
List of audit links if any:
Website Link: https://dirac.finance/
Logo (High resolution, will be shown with rounded borders): https://www.dirac.finance/black_icon.png
Current TVL:  5091,98 USDC 
Treasury Addresses (if the protocol has treasury)
Chain: Polygon ZKEVM
Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): 
Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): 
Short Description (to be shown on DefiLlama):
Onchain Derivatives Strategies. Earn yield through decentralized derivatives strategies, inspired by TradFi and adapted to DeFi.
Token address and ticker if any:
Category (full list at https://defillama.com/categories) *Please choose only one: Derivatives
Oracle Provider(s): Chainlink
Implementation Details: We use Chainlink to get on chain prices
Documentation/Proof:
forkedFrom (Does your project originate from another project):
methodology (what is being counted as tvl, how is tvl being calculated):
TVL is the value of usdc tokens locked in the protocol. The TVL calculation code does the below
Get the quantity of usdc.e/usdc tokens held by each vault.
Do the sum.
Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/dirac-finance exports TVL: 
        
 
 
    polygon_zkevm             4.01 k
    
    total                    4.01 k

**Comment by llamatester** [comment]
The adapter at projects/dirac-finance exports TVL: 
        
 
 
    polygon_zkevm             5.09 k
    
    total                    5.09 k

File: projects/dirac-finance/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+async function tvl(_, _1, _2, { api }) {
+    const collateralBalance1 = await api.call({
+        abi: 'erc20:balanceOf',
+        target: USDCE_CONTRACT,
+        params: [DIRAC_VAULT_1_CONTRACT],
+    });
+
+    const collateralBalance2 = await api.call({
+        abi: 'erc20:totalSupply',
+        target: DIRAC_VAULT_2_CONTRACT,
```
Comment (g1nt0ki) [inline projects/dirac-finance/index.js:18]: is there some way to fetch tvl on chain instead of counting supply as tvl?

Comment (salimdev04) [reply projects/dirac-finance/index.js:18]: To access the locked balance within these Vaults, it's the sole method available.

**Pull request merged**